provisioning/e2e/_provisioning_e2e.js

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+var Promise = require('bluebird');
+var pem = Promise.promisifyAll(require('pem'));
+var uuid = require('uuid');
+var assert = require('chai').assert;
+var debug = require('debug')('azure-device-provisioning-e2e');
+var Http = require('azure-iot-provisioning-device-http').Http;
+var ProvisioningServiceClient = require('azure-iot-provisioning-service').ProvisioningServiceClient;
+var Registry = require('azure-iothub').Registry;
+
+var idScope = process.env.IOTHUB_PROVISIONING_IDSCOPE;
+var provisioningConnectionString = process.env.IOTHUB_PROVISIONING_CONNECTION_STRING;
+var registryConnectionString = process.env.IOTHUB_CONNECTION_STRING;
+
+var provisioningTransports = [ Http ];
+
+describe('IoT Provisioning', function() {
+  var deviceId;
+  var registrationId;
+  this.timeout(30000);
+
+  var provisioningServiceClient = Promise.promisifyAll(ProvisioningServiceClient.fromConnectionString(provisioningConnectionString));
```

Comment:
>promisifyAll [](start = 42, length = 12)

I see the benefit but I'm worried we're adding one layer between the tests and the SDK that might alter the test results ?

Reply:
yeah, I go back and forth.  I was using this as an excuse to learn promises.  

---
In reply to: [148917838](https://github.com/Azure/azure-iot-sdk-node/pull/135#discussion_r148917838) [](ancestors = 148917838)

Reply:
for now, I'm going to leave it.  I'll change it later if we find a reason.

---
In reply to: [149802945](https://github.com/Azure/azure-iot-sdk-node/pull/135#discussion_r149802945) [](ancestors = 149802945,148917838)